### PR TITLE
Mock testing sendEmail function

### PR DESCRIPTION
**Pull request by sandlerben**

### CODE REVIEW

**Comment by yoninachmany** [comment]
@sandlerben insight on why checks are failing?

**Comment by sandlerben** [comment]
@yoninachmany trying very hard to figure that out right now

**Comment by sandlerben** [comment]
It's working now and I'm not sure what the problem was :/

File: transcription/transcription_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package transcription
+
+import (
+	"errors"
+	"net/smtp" // mock
```
Comment (yoninachmany) [inline transcription/transcription_test.go:5]: what does "// mock" comment mean?

**Comment by sandlerben** [comment]
r @yoninachmany 

Thanks for the comment. Feel free to review the rest!

File: transcription/transcription_test.go (hunk 1 above)
Comment (sandlerben) [inline transcription/transcription_test.go:5]: This is how `withmock` works. It scans the imports for the `// mock` comment to determine which packages to mock. Do you think I should document this better?

File: transcription/transcription_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
```
Comment (yoninachmany) [inline transcription/transcription_test.go:12]: `username, password, to` values here are dummy - is that ok or will that mean that email gets rejected?

File: transcription/transcription_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	// Setup the mock package
+	smtp.MOCK().SetController(ctrl)
+
+	gomock.InOrder(
+		smtp.EXPECT().PlainAuth("", username, password, "smtp.gmail.com"),
+		smtp.EXPECT().SendMail(addr, gomock.Any(), username, to, gomock.Any()).Return(errors.New("Bad!")),
+	)
+
+	err := SendEmail(username, password, host, port, to, subject, body)
```
Comment (yoninachmany) [inline transcription/transcription_test.go:53]: why will this lead to error if it's passed same params as call above, which should work?

File: transcription/transcription_test.go (hunk 2 above)
Comment (sandlerben) [inline transcription/transcription_test.go:12]: The whole point of mocking is that the email _cannot_ get rejected! We should talk about mocking at some point. Maybe I can go over it on Saturday.

File: transcription/transcription_test.go (hunk 3 above)
Comment (sandlerben) [inline transcription/transcription_test.go:53]: Look at line 50. I mock the SendMail function here to always return an error.

**Comment by sandlerben** [comment]
@yoninachmany make sure you understand the why and how of test mocking before giving this review a :+1:

File: transcription/transcription_test.go (hunk 1 above)
Comment (yoninachmany) [inline transcription/transcription_test.go:5]: no sorry i didnt look up the mocking package that you used before

File: transcription/transcription_test.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	subject  = "subject"
+	body     = "body"
+)
+
+func TestSendEmail(t *testing.T) {
+	assert := assert.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Setup the mock package
```
Comment (yoninachmany) [inline transcription/transcription_test.go:28]: // Setup the smtp mock package

**Comment by yoninachmany** [comment]
@sandlerben ok i get mocking now. did you get these tests to pass?

**Comment by sandlerben** [comment]
Yes! Also check travis to see the test results.

**Comment by yoninachmany** [comment]
@sandlerben please pull in latest version of `send-email`

**Pull request closed without merging**